onnx/test/hub_test.py

```diff
@@ -107,6 +107,15 @@ def test_download_model_with_test_data(self) -> None:
             member="test_data_set_0", container=files, msg="Test data not found"
         )
 
+    def test_download_model_with_test_data_tarball_extraction(self) -> None:
+        self.assertRaisesRegex(
+            AssertionError,
+            "The tarball file in downloading model contains .*harmful payload.$",
+            lambda: hub.download_model_with_test_data(
+                "mnist", repo="sunriseXu/onnx", force_reload=True, silent=True
```

Comment:
Avoid dependency on external files in tests

Reply:
Thanks for your review. The function `download_model_with_test_data` needs external repo with malicious tarball to trigger the assert error. I tested locally, and passed the test. I can remove this test in case of security issue.